
# Question 2.3

1. Exons are under the highest negative selection, as SNP enrichment decreases the most as minor allele frequency increases. 

2. SNP enrichment generally decreases as MAF increases. This is because increasing MAF represents increasing variability of that SNP (more presence of the minor allele). Thus, if sequence specificity is needed (as in the case of exons), these regions are less tolerant of SNPs and are thus less enriched. 

3. The relative levels of enrichment with respect to MAF make sense. Across the range of MAF values, introns have high SNP enrichment and are thus tolerant of variablility, which makes sense as these regions are spliced out and dont code for anything (the only important regions are splice sites, which does explain the slight negative correlation). cCREs also have relatively high SNP enrichment across the board, but  still decrease more than introns as MAF increases, suggesting they are less tolerant of SNP variability than introns. This makes sense as regulatory elements still perform important functions with respect to gene expression. Exons are the least tolerant, with the largest decrease in enrichment with increasing MAF. "Other" regions of the genome have the opposite trend, and increase in SNP enrichment as MAF increases. This is likely because these regions are noncoding regions where variability does not significantly affect function. 




